Editor/Drawers/ReferenceDrawer.cs

```diff
@@ -219,12 +219,11 @@ protected void SetModeValue(SerializedProperty property, ReferenceMode mode)
 
         protected object GetRawReferenceValue(SerializedProperty property)
         {
-#if UNITY_2021_1_OR_NEWER
+#if UNITY_2021_2_OR_NEWER
             return property.RawReferenceProperty().managedReferenceValue;
 #else
-                ISerializableInterface instance =
-                    (ISerializableInterface)FieldInfo.GetValue(property.serializedObject.targetObject);
-                return instance.GetRawReference();
+            var target = (ISerializableInterface) SerializedPropertiesUtils.GetTargetObjectOfProperty(property);
+            return target.GetRawReference();
```

Comment:
Is there a good reason for this change, or would this work in its original form as well?

Reply:
Hi, yes, there is a reason. It solves a different issue, sorry for adding it to the same pull request. The issue is that it raises an error when you add managed class (raw reference) to a list.

Steps to reproduce:
1. Create a new project and install your package (of course in Unity 2021.1 or a previous one)
2. Create an interface: let's name it IMyInterface.
3. Create a serializable class MyClass like:
```
[Serializable]
public class MyClass : IMyInterface
{
}
```
4. Create a MonoBehaviour with a list like:
```
public class Test : MonoBehaviour
{
    [SerializeField] private List<SerializableInterface<IMyInterface>> _list;
}
```
5. Add it to an object in a scene
6. And try to add an element to the list of type MyClass
7. You should get the following error:

UnityEditor.PropertyDrawer.OnGUISafe (UnityEngine.Rect position, UnityEditor.SerializedProperty property, UnityEngine.GUIContent label) (at <42425271a85c48a781b5ae89b9328419>:0)
UnityEditor.PropertyHandler.OnGUI (UnityEngine.Rect position, UnityEditor.SerializedProperty property, UnityEngine.GUIContent label, System.Boolean includeChildren, UnityEngine.Rect visibleArea) (at <42425271a85c48a781b5ae89b9328419>:0)
UnityEditor.PropertyHandler.OnGUI (UnityEngine.Rect position, UnityEditor.SerializedProperty property, UnityEngine.GUIContent label, System.Boolean includeChildren) (at <42425271a85c48a781b5ae89b9328419>:0)
UnityEditorInternal.ReorderableList+Defaults.DrawElement (UnityEngine.Rect rect, UnityEditor.SerializedProperty element, System.Object listItem, System.Boolean selected, System.Boolean focused, System.Boolean draggable, System.Boolean editable) (at <42425271a85c48a781b5ae89b9328419>:0)
UnityEditorInternal.ReorderableList.DoListElements (UnityEngine.Rect listRect, UnityEngine.Rect visibleRect) (at <42425271a85c48a781b5ae89b9328419>:0)
UnityEditorInternal.ReorderableList.DoList (UnityEngine.Rect rect, UnityEngine.Rect visibleRect) (at <42425271a85c48a781b5ae89b9328419>:0)
UnityEditorInternal.ReorderableListWrapper.Draw (UnityEngine.GUIContent label, UnityEngine.Rect r, UnityEngine.Rect visibleArea, System.Boolean includeChildren) (at <42425271a85c48a781b5ae89b9328419>:0)
UnityEditor.PropertyHandler.OnGUI (UnityEngine.Rect position, UnityEditor.SerializedProperty property, UnityEngine.GUIContent label, System.Boolean includeChildren, UnityEngine.Rect visibleArea) (at <42425271a85c48a781b5ae89b9328419>:0)
UnityEditor.GenericInspector.OnOptimizedInspectorGUI (UnityEngine.Rect contentRect) (at <42425271a85c48a781b5ae89b9328419>:0)
UnityEditor.UIElements.InspectorElement+<>c__DisplayClass59_0.<CreateIMGUIInspectorFromEditor>b__0 () (at <562725784adc4f679865dd6c5d4174e9>:0)
UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr, Boolean&)

The problem is that in that line (where the error comes out, see the image below) FieldInfo is pointing to the list field, not to the specific element you are adding. So, the (ISerializableInterface) cast will fail.
![Fix issue #51 by mnicolas94 · Pull Request #52 · ThundernerdUnity3D-SerializableInterface - Google Chrome](https://user-images.githubusercontent.com/35781652/188331828-48996fe9-a19f-4f4c-84d8-fbe1d81ccd85.jpg)

Reply:
I'm not a huge fan of the way the SerializedPropertiesUtils class is set up, so I'll dive in myself to make it to my liking. Once that has been done this should be good to merge :)

Reply:
Ok, seems good to me. Im not a fan either, I just use it because it works. So, im looking foraward your solution.

P.S: really great package, thanks for making it

Reply:
Alright, I've added a new class named `SerializedPropertyUtilities` to the project. You can call the `GetValue` method from there. So if you could remove the added file from your PR that would be great :)